src/ndv/viewer/_backends/_vispy.py

```diff
@@ -327,6 +302,40 @@ def border_color(self, color: cmap.Color | None = None) -> None:
     def remove(self) -> None:
         self._roi.parent = None
 
+    def cursor_at(self, pos_xy: tuple[float, float]) -> Qt.CursorShape | None:
```

Comment:
this could alternatively return a flag enum, that the viewer would then use to decide the cursor:

```python
class RoiLocation(enum.Flag):
    TOP_EDGE = enum.auto()
    BOTTOM_EDGE = enum.auto()
    LEFT_EDGE = enum.auto()
    RIGHT_EDGE = enum.auto()
    INSIDE = enum.auto()
    OUTSIDE = enum.auto()
    TOP_LEFT_CORNER = TOP_EDGE | LEFT_EDGE
    TOP_RIGHT_CORNER = TOP_EDGE | RIGHT_EDGE
    BOTTOM_LEFT_CORNER = BOTTOM_EDGE | LEFT_EDGE
    BOTTOM_RIGHT_CORNER = BOTTOM_EDGE | RIGHT_EDGE
```

and then:

```python
def position_at(self, pos_xy: tuple[float, float]) -> RoiLocation:
```

and then:

if `position_at(some_pos) & RoiLocation.TOP_LEFT_CORNER` ...